exporter.go

```diff
@@ -408,5 +416,49 @@ func scrapeMySQLConnectionList(db *sql.DB, ch chan<- prometheus.Metric) error {
 	return rows.Err()
 }
 
+const detailedMySQLProcessListQuery = "SELECT user, db, cli_host, hostgroup, COUNT(*) as count from stats_mysql_processlist group by user, db, cli_host, hostgroup"
+
+var detailedMySQLProcessListMetrics = map[string]*metric{
+	"detailed_connection_count": {name: "detailed_client_connection_count", valueType: prometheus.GaugeValue, help: "Number of client connections per user, db, host and hostgroup."},
+}
+
+type processListResult struct {
+	user, db, clientHost, hostGroup string
+	count                           float64
+}
+
+func scrapeDetailedMySQLConnectionList(db *sql.DB, ch chan<- prometheus.Metric) error {
+	rows, err := db.Query(detailedMySQLProcessListQuery)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var res processListResult
+
+		err := rows.Scan(&res.user, &res.db, &res.clientHost, &res.hostGroup, &res.count)
+		if err != nil {
+			return err
+		}
+
+		m := detailedMySQLProcessListMetrics["detailed_connection_count"]
+
+		ch <- prometheus.MustNewConstMetric(
+			prometheus.NewDesc(
+				prometheus.BuildFQName(namespace, "processlist", m.name),
+				m.help,
+				[]string{"user", "db", "client_host", "hostgroup"},
+				nil,
+			),
+			m.valueType,
+			res.count,
+			res.user, res.db, res.clientHost, res.hostGroup,
+		)
+	}
```

Comment:
We should check `rows.Err()` and log an error.

Reply:
@OGKevin ^

Reply:
https://github.com/percona/proxysql_exporter/pull/25/files#diff-557eb2a359922e8de5f18397fed0cd99R461 👁 

it this not what you meant ?

Reply:
Sorry, missed that :)

Reply:
np